### PR TITLE
Add test target with qemu

### PR DESCRIPTION
**Pull request by jsarzy**

### CODE REVIEW

File: build/Dockerfile [hunk 1]
```diff
@@ -78,3 +78,26 @@ ENV PATH="/opt/phoenix/arm-phoenix/bin:/opt/phoenix/i386-pc-phoenix/bin:/opt/pho
...
+        python3-pip \
+        qemu-system-i386 \
+        qemu-system-misc \
+        opensbi \
+        u-boot-qemu \
+        qemu-utils \
+    && rm -rf /var/lib/apt/lists/*
+
+# get python requirements
+RUN pip3 install \
```
Comment (jsarzy) [inline build/Dockerfile:97]: Here we would like to get requirements.txt from phoenix-rtos-project or phoenix-rtos-tests repository but [PR from phoenix-rtos-test](https://github.com/phoenix-rtos/phoenix-rtos-tests/pull/3) is not merged yet.

File: build/Dockerfile [hunk 2]
```diff
@@ -78,3 +78,26 @@ ENV PATH="/opt/phoenix/arm-phoenix/bin:/opt/phoenix/i386-pc-phoenix/bin:/opt/pho
...
+        pyserial \
+        pyyaml \
+        pexpect
+
+ENTRYPOINT ["/bin/bash", "-l", "-c"]
```
Comment (jsarzy) [inline build/Dockerfile:102]: Same as above.
`ENTRYPOINT ["python3", "phoenix-rtos-tests/runner.py"]`

File: build/Dockerfile (hunk 1 above)
Comment (nalajcie) [reply build/Dockerfile:97]: 1. add `--no-cache-dir` so we would not include cached packets in the final image (as per
2. Regarding the requirements.txt - I honestly don't know what the best approach would be:
   - either we would need to remember to rebuild the image each time `phoenix-rtos-test/requirements.txt` would change
   -  or list the requirements here directly and keep them in sync with `requirements.txt`.

As both solutions would result in early failure, I'm ok with both solutions.

File: build/Dockerfile (hunk 2 above)
Comment (nalajcie) [reply build/Dockerfile:102]: docker image can only have one entry point, please use separate file.

**Review by nalajcie — CHANGES_REQUESTED** [review]

**Review by nalajcie — COMMENTED** [review]
Nice, after some thinking - can You put it in the `devel` subdirectory, as I would like to name the image `phoenixrtos/devel` (`dev` can be also a shortcut for `device`, I would like to avoid the confusion ;)).

**Comment by nalajcie** [comment]
FYI, I've built and pushed the docker image to DockerHub: https://hub.docker.com/r/phoenixrtos/devel/tags?page=1&ordering=last_updated

**Review by nalajcie — APPROVED** [review]

**Review by nalajcie — COMMENTED** [review]
edit: please add JIRA ticket ID

**Review by nalajcie — APPROVED** [review]

**Pull request merged**